### PR TITLE
[DOC] Enhanced RDoc for String

### PR DESCRIPTION
**Pull request by BurdetteLamar**
- Adds to doc for String.new, also making it compliant with documentation_guide.rdoc.
- Fixes some broken links in io.c (that I failed to correct yesterday).

### CODE REVIEW

File: doc/string/new.rdoc [hunk 1]
```diff
@@ -1,36 +1,51 @@
...
-which may affect performance:
-  String.new(capacity: 1) # => ""
-  String.new(capacity: 4096) # => ""
+With optional argument +string+ and no keyword arguments,
+returns a copy of +string+ with the same encoding:
+
+  String.new('foo')               # => "foo"
+  String.new('Que veut dire ça?') # => "Que veut dire ça?"
+  String.new('тест')              # => "тест"
+  String.new('こんにちは')          # => "こんにちは"
```
Comment (peterzhu2118) [inline doc/string/new.rdoc:15]: I don't think we need so many examples of this. I think two is enough (one with only ASCII characters and one with non-ASCII characters).

**Review by peterzhu2118 — COMMENTED** [review]
One minor comment. Looks like there are conflicts that are preventing merge?

File: doc/string/new.rdoc (hunk 1 above)
Comment (BurdetteLamar) [reply doc/string/new.rdoc:15]: We've been doing ascii, Cyrillic, and Japanese, and I'm content to continue with that.  The French was here already; I'll delete it if you like.

**Comment by BurdetteLamar** [comment]
> One minor comment. Looks like there are conflicts that are preventing merge?

I'm not seeing conflicts.  I fixed a bunch earlier.

**Comment by peterzhu2118** [comment]
GitHub is showing conflicts for me.

image

**Comment by BurdetteLamar** [comment]
My page sez:  This branch has no conflicts with the base branch
Merging can be performed automatically. 

Lemme do the next push (remove French), and we'll see what's then.

**Comment by BurdetteLamar** [comment]
@peterzhu2118, how now?

**Review by peterzhu2118 — APPROVED** [review]
Looks good!

**Comment by nobu** [comment]
I think `Encoding@` is better because the class name will not change.

**Comment by BurdetteLamar** [comment]
> I think `Encoding@` is better because the class name will not change.

I meant for the topic of the encodings.rdoc page to be the concept encodings, not the class Encoding.

**Pull request merged**